lib/python/qmk/cli/cformat.py

```diff
@@ -0,0 +1,34 @@
+"""Format C code according to QMK's style.
+"""
+import os
+import subprocess
+
+from milc import cli
+
+
+@cli.argument('files', nargs='*', help='Filename(s) to format.')
+@cli.entrypoint("Format C code according to QMK's style.")
+def main(cli):
+    """Format C code according to QMK's style.
+    """
+    clang_format = ['clang-format', '-i']
+
+    # Find the list of files to format
+    if not cli.args.files:
+        for dir in ['drivers', 'quantum', 'tests', 'tmk_core']:
+            for dirpath, dirnames, filenames in os.walk(dir):
+                if 'tmk_core/protocol/usb_hid' in dirpath:
+                    continue
+
+                for name in filenames:
+                    if name.endswith('.c') or name.endswith('.h') or name.endswith('.cpp'):
+                        cli.args.files.append(os.path.join(dirpath, name))
+
+    # Run clang-format on the files we've found
+    try:
+        subprocess.run(clang_format + cli.args.files, check=True)
```

Comment:
is there any limit to how many files you can specify to a subprocess?

Reply:
Yes, but that limit is very large, 2097152 items on linux. Per POSIX the minimum number that can be as low as 4096 but every modern system has made that much larger by now.